src/core.rs

```diff
@@ -769,33 +787,29 @@ impl<T> Webauthn<T> {
+        let auth_data = self.verify_credential_internal(rsp, policy, chal.into(), &cred)?;
+        let counter = auth_data.counter;
 
         // If the signature counter value authData.signCount is nonzero or the value stored in
         // conjunction with credential’s id attribute is nonzero, then run the following sub-step:
-        if counter > 0 {
-            // If the signature counter value authData.signCount is
-            if counter > cred.counter {
-                // greater than the signature counter value stored in conjunction with credential’s id attribute.
-                //       Update the stored signature counter value, associated with credential’s id attribute,
-                //       to be the value of authData.signCount.
-                Ok(Some((cred.cred_id, counter)))
-            // If all the above steps are successful, continue with the authentication ceremony as
-            // appropriate. Otherwise, fail the authentication ceremony.
-            } else {
-                // less than or equal to the signature counter value stored in conjunction with credential’s id attribute.
-                //      This is a signal that the authenticator may be cloned, i.e. at least two copies
-                //      of the credential private key may exist and are being used in parallel. Relying
-                //      Parties should incorporate this information into their risk scoring. Whether the
-                //      Relying Party updates the stored signature counter value in this case, or not,
-                //      or fails the authentication ceremony or not, is Relying Party-specific.
-                Err(WebauthnError::CredentialPossibleCompromise)
-            }
-        } else {
-            // If all the above steps are successful, continue with the authentication ceremony as
-            // appropriate. Otherwise, fail the authentication ceremony.
-            Ok(None)
+        let check_counter = counter > 0 || cred.counter > 0;
+
+        // greater than the signature counter value stored in conjunction with credential’s id attribute.
+        //       Update the stored signature counter value, associated with credential’s id attribute,
+        //       to be the value of authData.signCount.
+        // less than or equal to the signature counter value stored in conjunction with credential’s id attribute.
+        //      This is a signal that the authenticator may be cloned, i.e. at least two copies
+        //      of the credential private key may exist and are being used in parallel. Relying
+        //      Parties should incorporate this information into their risk scoring. Whether the
+        //      Relying Party updates the stored signature counter value in this case, or not,
+        //      or fails the authentication ceremony or not, is Relying Party-specific.
+        let counter_shows_compromise = auth_data.counter <= cred.counter;
```

Comment:
nit pick, we could do this inside of an 'if check_counter', but that's a super minor point.